main/IpsDisplay.cpp

```diff
@@ -2298,8 +2315,40 @@ void IpsDisplay::drawRetroDisplay( int airspeed_kmh, float te_ms, float ate_ms,
 		drawAvg( acl, acl-average_climbf );
 		average_climbf = acl;
 	}
+
+	// Check if the volume has changed.
+	static auto lastAudioVolume = audio_volume.get();
+	static auto lastVolumeChangeTime_ms = 0;
+	static bool shouldClearDisplay = false;
+	static bool shouldDrawVolume = false;
+	if (lastAudioVolume != audio_volume.get()) {
+		// Update the helper variables
+		lastAudioVolume = audio_volume.get();
+		lastVolumeChangeTime_ms = millis();
+		shouldDrawVolume = true;
+	}
+
+	//  Check if the timer has expired. If it has, check if we should clear the display
+	static uint16_t volumeDisplayPersistance_ms = 1500;
+	if (shouldDrawVolume == true) {
+		if (millis() - lastVolumeChangeTime_ms < volumeDisplayPersistance_ms) {
+			// Draw the new volume
+			drawVolume(220, 180, audio_volume.get());
+		} else {
+			// Set flags
+			shouldDrawVolume = false;
+			shouldClearDisplay = true;
+		}
+	}
+
 	// ESP_LOGI(FNAME,"IpsDisplay::drawRetroDisplay  TE=%0.1f  x0:%d y0:%d x2:%d y2:%d", te, x0, y0, x2,y2 );
 	xSemaphoreGive(spiMutex);
+
+	// Check if necessary to clear display
+	if (shouldClearDisplay == true) {
+		shouldClearDisplay = false;
```

Comment:
This looks and feels wrong. Is there a better way to make the drawn text simply disappear?

Reply:
There is unfortunately no way. An overlay text is not supported with this graphics protocol.